beater/api/mux.go

```diff
@@ -312,6 +326,16 @@ func rootMiddleware(cfg *config.Config, authenticator *auth.Authenticator) []mid
 	)
 }
 
+func firehoseMiddleware(cfg *config.Config, authenticator *auth.Authenticator, ratelimitStore *ratelimit.Store, m map[request.ResultID]*monitoring.Int) []middleware.Middleware {
+	backendMiddleware := append(apmMiddleware(m),
+		middleware.ResponseHeadersMiddleware(cfg.ResponseHeaders),
+		middleware.FirehoseAuthMiddleware(),
+		middleware.AuthMiddleware(authenticator, true),
+		middleware.AnonymousRateLimitMiddleware(ratelimitStore),
```

Comment:
```suggestion
```

- Auth middleware isn't relevant, since you're using the authenticator directly in the handler.
- Anonymous auth isn't relevant to Firehose